src/location-picker/location-picker.component.tsx

```diff
@@ -55,8 +56,16 @@ interface LocationPickerProps {
 }
 
 const LocationPicker: React.FC<LocationPickerProps> = (props) => {
+  const userDefaultLoginLocation: string = "userDefaultLoginLocationKey";
+  const getDefaultUserLoginLocation = (): string => {
+    return window.localStorage.getItem(
+      `${userDefaultLoginLocation}${props.currentUser}`
+    );
+  };
   const [locationData, setLocationData] = React.useState<LocationDataState>({
-    activeLocation: "",
+    activeLocation: getDefaultUserLoginLocation()
```

Comment:
Rather use `getDefaultUserLoginLocation() ?? ''`.

On the other hand, this is a bit fishy as `localStorage` may get corrupted or the available locations may change. Thus we definitely will need to check if the location is valid.

Reply:
I have added a check for validation for the login locations.